uaclient/config.py

```diff
@@ -186,6 +190,50 @@ def apt_https_proxy(self, value: str):
         self.cfg["ua_config"]["apt_https_proxy"] = value
         self.write_cfg()
 
+    @property
+    def ua_apt_http_proxy(self) -> Optional[str]:
+        return self.cfg.get("ua_config", {}).get("ua_apt_http_proxy")
+
+    @ua_apt_http_proxy.setter
+    def ua_apt_http_proxy(self, value: str):
+        if "ua_config" not in self.cfg:
+            self.cfg["ua_config"] = {}
+        self.cfg["ua_config"]["ua_apt_http_proxy"] = value
+        self.write_cfg()
+
+    @property
+    def ua_apt_https_proxy(self) -> Optional[str]:
+        return self.cfg.get("ua_config", {}).get("ua_apt_https_proxy")
+
+    @ua_apt_https_proxy.setter
+    def ua_apt_https_proxy(self, value: str):
+        if "ua_config" not in self.cfg:
+            self.cfg["ua_config"] = {}
+        self.cfg["ua_config"]["ua_apt_https_proxy"] = value
+        self.write_cfg()
+
+    @property
+    def global_apt_http_proxy(self) -> Optional[str]:
+        return self.cfg.get("ua_config", {}).get("global_apt_http_proxy")
+
+    @global_apt_http_proxy.setter
+    def global_apt_http_proxy(self, value: str):
+        if "ua_config" not in self.cfg:
+            self.cfg["ua_config"] = {}
+        self.cfg["ua_config"]["global_apt_http_proxy"] = value
+        self.write_cfg()
+
+    @property
+    def global_apt_https_proxy(self) -> Optional[str]:
+        return self.cfg.get("ua_config", {}).get("global_apt_https_proxy")
+
+    @global_apt_https_proxy.setter
+    def global_apt_https_proxy(self, value: str):
+        if "ua_config" not in self.cfg:
+            self.cfg["ua_config"] = {}
+        self.cfg["ua_config"]["global_apt_https_proxy"] = value
+        self.write_cfg()
```

Comment:
I understand the first point, where we encourage people to use the `global_*` property by giving them a warning if they still use the older `apt_*` property and not `global_*`.
On the second point, do you mean we have the `apt_*` available on the cli options but the uaconfig internals should not have the property? If that is what you mean, then it is a better approach, I see no issue with it. It will still keep maintain cli api for configuring the apt_* proxy and reduce the properties of UAConfig.